static/js/controllers/messages-content.js

```diff
@@ -93,9 +105,8 @@ angular.module('inboxControllers').controller('MessagesContentCtrl',
           $scope.firstUnread = _.min(unread, function(message) {
             return message.doc.reported_date;
           });
-          $scope.selected.contact = contactModel.doc ? contactModel : { name: id };
+          $scope.selected.contact = contactModel;
           $scope.selected.messages = conversation;
-          $scope.setTitle((contactModel.doc && contactModel.doc.name) || id);
```

Comment:
You still need to call `setTitle` somewhere - this is what sets the title bar for mobile.